src/Schema/migrations/index.js

```diff
@@ -83,51 +81,54 @@ export type SchemaMigrations = $Exact<{
-      'Current schema version must be greater than minimum migrable version',
-    )
-
     invariant(Array.isArray(migrations), 'Missing migrations array')
 
     // validate migrations format
     migrations.forEach(migration => {
       invariant(isObject(migration), `Invalid migration (not an object) in schema migrations`)
-      const { from, to, steps } = migration
+      const { toVersion, steps } = migration
+      invariant(typeof toVersion === 'number', 'Invalid migration - `toVersion` must be a number')
       invariant(
-        typeof from === 'number' && typeof to === 'number' && to > from,
-        'Invalid migration - `to` version must be greater than `from` version',
+        toVersion >= 2,
+        `Invalid migration to version ${toVersion}. Minimum possible migration version is 2`,
       )
       invariant(
         Array.isArray(steps) && steps.every(step => typeof step.type === 'string'),
-        `Invalid migration steps for migration from version ${from} to ${to}. 'changes' should be an array of migration step calls`,
+        `Invalid migration steps for migration to version ${toVersion}. 'steps' should be an array of migration step calls`,
       )
     })
 
-    // validate if migration spec actually covers every schema version it says it supports
-    const chronologicalMigrations = [...migrations].reverse()
-    let maxCoveredRange = minimumVersion
+    // validate that migration spec is reverse-chronological and without gaps
+    let maxCoveredVersion: ?number = null
 
-    chronologicalMigrations.forEach(({ from, to }) => {
-      invariant(
-        maxCoveredRange === from,
-        `Invalid migrations! schemaMigrations() says it covers schema versions from ${minimumVersion} to ${currentVersion}, but there is no listed migration from ${maxCoveredRange} to ${from}. Remember that migrations must be listed in reverse chronological order`,
-      )
-      maxCoveredRange = to
+    migrations.forEach(migration => {
+      const { toVersion } = migration
+      if (maxCoveredVersion) {
+        invariant(
+          toVersion === maxCoveredVersion - 1,
+          `Invalid migrations! Migration ${JSON.stringify(
+            migration,
+          )} is to version ${toVersion}, but previously listed migration is to version ${maxCoveredVersion}. Remember that migrations must be listed in reverse chronological order and without gaps -- migration to newest version must be at the top, and every following migration must be to version 1 number smaller`,
+        )
+      }
+      maxCoveredVersion = toVersion
     })
-    invariant(
-      maxCoveredRange === currentVersion,
-      `Invalid migrations! schemaMigrations() says the current version is ${currentVersion}, but migrations listed only cover schema versions range from ${minimumVersion} to ${maxCoveredRange}. Remember that migrations must be listed in reverse chronological order`,
-    )
   }
+
+  const sortedMigrations = sortMigrations(migrations)
```

Comment:
Why migrations have to be in a reverse chronological order if we are sorting it anyway? We can first sort it and then check for completeness.

Reply:
yeah, artifact of previous implementation. If you think this is going to be annoying, I can change it.

Reply:
I don't think so. Most people will do it anyway :) But maybe we don't have to force reverse order? Some users might find chronological order better.

Reply:
OK, fixed